packages/devextreme/js/__internal/grids/grid_core/editing/m_editing.ts

```diff
@@ -780,12 +781,8 @@ class EditingControllerImpl extends modules.ViewController {
 
   _addInsertInfo(change, parentKey?) {
     let insertInfo;
-    let { key } = change;
-
-    if (!isDefined(key)) {
-      key = String(new Guid());
-      change.key = key;
-    }
+    this.ensureChangeKeyValue(change);
```

Comment:
I would suggest that **ensureChangeKeyValue** method doesn't modify argument, but return a value

`change.key = this.createKeyValue(change);`